### ISSUE TITLE
Google Drive support on Android?

### ISSUE THREAD

**Issue by rotter** [yasirkula/UnityNativeGallery#41]
My game was originally made on native Android, and my gallery integration was a bit messy but it would allow the user to select photos from his Google Drive account (besides local photos).

Now I migrated my game to Unity and found your awesome plugin which saved me a lot of time, but some players are complaining that they lost the ability to read the photos from their Google Drive account.

So here is my suggestion, and I believe a lot of your users would be happy with this feature as well.

Thanks again for such a great plugin!

**Comment by rotter** [comment]
Obs: I can provide the code for my old gallery integration if it helps you, but as far as I remember it's very messy, and I dont think I will be able to pinpoint to you which part of the code makes Google Drive work (I dont remember the code anymore, it's probably around 3 years old)

**Comment by yasirkula** [comment]
I think this can easily be done by changing [these lines](https://github.com/yasirkula/UnityNativeGallery/blob/d16ff30c1342d5baab1295dc10736bb79dde35cf/JAR%20Source/NativeGalleryMediaPickerFragment.java#L69-L72) like this:

```java
intent = new Intent( Intent.ACTION_GET_CONTENT );
intent.addCategory( Intent.CATEGORY_OPENABLE );
```

_addCategory_ line may be redundant.

**Comment by yasirkula** [comment]
The only thumbnail issue I saw so far in Android is when you try to capture an image with camera. I've created [a plugin](https://github.com/yasirkula/UnityNativeCamera/) that doesn't seem to have this issue but it only works with camera.

**Comment by rotter** [comment]
From what I remember this thumbnail issue happened when selecting photos that are not on the device (ie, from any cloud service). Maybe that's why I tried putting that flag "LOCAL_ONLY", but I don't remember much, and I dont think I ever fully fixed that thumbnail problem.

I remember I tried some other hacks as well, that's why I said my code is messy. From what Im seeing in my code I used this proposed solution:
https://stackoverflow.com/questions/20067508/get-real-path-from-uri-android-kitkat-new-storage-access-framework/20559175#20559175

Anyway if this thumbnail problem still exists (maybe it was fixed by newer Android versions) I think I can put 2 buttons on my game to open the gallery, one for local files (which calls the current code) and a new button for cloud files that calls this new modified code that might work or not (if it doesn't work well the player can still use the local files button)
